src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -71,20 +72,30 @@ public sealed partial class TextLoader
             char separator = '\t', bool allowQuotedStrings = true,
             bool supportSparse = true, bool trimWhitespace = false)
         {
-            var fields = typeof(TInput).GetFields();
-            Arguments.Column = new TextLoaderColumn[fields.Length];
-            for (int index = 0; index < fields.Length; index++)
+            var userType = typeof(TInput);
+
+            var fieldInfos = userType.GetFields();
```

Comment:
Should we be checking for public and instance fields here?  Just like we are in GetProperties.